utils/scarb-ui/src/args/packages_filter.rs

```diff
@@ -99,6 +102,30 @@ impl PackagesFilter {
         Self::do_match_all::<S>(specs, self.workspace, members)
     }
 
+    /// Generate a new [`PackagesFilter`] for the given slice  of packages.
+    ///
+    /// This is useful when you want to build an env filter from matched packages.
+    /// See [`PackagesFilter::to_env`] for more details.
+    pub fn generate_for<S: PackagesSource>(packages: &[S::Package]) -> Self {
+        let names: Vec<String> = packages
+            .iter()
+            .map(|p| S::package_name_of(p).to_string())
+            .collect();
+        Self {
+            package: names,
+            workspace: false,
+        }
+    }
+
+    /// Get the packages filter as an OsString.
+    ///
+    /// This value can be passed as `SCARB_PACKAGES_FILTER` variable to child processes.
+    pub fn to_env(self) -> OsString {
```

Comment:
Hmm, https://doc.rust-lang.org/stable/std/process/struct.Command.html#method.env accepts anything `AsRef<OsStr>`, so this means, we could return plain `String` here, and that API could still be ergonomic in main case.

Reply:
Yes, we could, OsString emphasizes the point of this function though - I am not sure if we should encourage parsing this string for other use cases 🤔

Reply:
True, let's leave this as-is then